src/io/dataset_loader.cpp

```diff
@@ -1303,6 +1444,7 @@ void DatasetLoader::ExtractFeaturesFromFile(const char* filename, const Parser*
           }
         }
       }
+      // TODO(shiyu1994): should be start_idx + i ?
```

Comment:
still todo?

Reply:
This should be `start_idx + i`, and it has been a potential bug since very long time ago. In line 1384, we use `PushData(tid, sub_feature, start_idx + i...` in the previous few lines. So using `dataset->FinishOneRow(tid, i,` here is an obvious inconsistence. The true line index is `start + i`, instead of `i`.

This code is not changed in PR, it is inherited from very early version. And it has long been a potential bug.